### ISSUE TITLE
Wrong detection of Squiz.Arrays.ArrayDeclaration.KeyNotAligned when key contains space

### ISSUE THREAD

**Issue by adriankirchner** [squizlabs/PHP_CodeSniffer#433]
Here are some more details:
```
$ phpcs --version
PHP_CodeSniffer version 2.1.0 (stable) by Squiz (http://www.squiz.net)
```
content of test.php
```php
<?php

$foo = array(
        $this->getViewName() . '.id'    => 'value',
        $this->getViewName() . '.title' => 'value',
       );
```
content of ruleset.xml
```xml
<?xml version="1.0"?>
<ruleset name="test">
    <description>test ruleset</description>
    <rule ref="Squiz.Arrays.ArrayDeclaration"/>
</ruleset>
```
executing this setup results in the following errors:
```
$ phpcs -pv --standard=ruleset.xml test.php
Registering sniffs in the test standard... DONE (1 sniffs registered)
Creating file list... DONE (1 files in queue)
Changing into directory /path/to/
Processing test.php [PHP => 44 tokens in 6 lines]... DONE in 7ms (2 errors, 0 warnings)

FILE: /path/to/test.php
--------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
--------------------------------------------------------------------------------
 4 | ERROR | [x] Array key not aligned correctly; expected 8 spaces but found
   |       |     31
 5 | ERROR | [x] Array key not aligned correctly; expected 8 spaces but found
   |       |     31
--------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------

Time: 12ms; Memory: 2Mb
```
Please ping me if you need more information.

**Comment by gsherwood** [comment]
Thanks for the really clear bug report. The sniff assumes there are no spaces in the array index, hence the error.

**Comment by adriankirchner** [comment]
Ah, would've never thought that since spaces are possible in array indexes:
```php
<?php
$a = array('a b' => 'c');
var_dump($a['a b']);
```
```
$ php test.php 
string(1) "c"
```
The error description `[x] Array key not aligned correctly; expected 8 spaces but found 31` states, that phpcs expects the beginning of the index declaration at char 8 (where the method call begins with `$this`) but found the supposed beginning at char 31 where the actual string declaration after the concatenation begins.
But I really don't know the internals of phpcs, therefore pure speculation from my side ;)

**Comment by gsherwood** [comment]
I've fixed this up now. Thanks again for the report and please let me know if you have any more test cases that are still breaking.

**Comment by adriankirchner** [comment]
thanks a lot, this was fast! It works perfectly

**Comment by gsherwood** [comment]
I just found another issue where it breaks if you have comments between array items. I've just pushed a fix for this as well. If you come across that, just update again.
